### DISCUSSION TITLE
Support unknown files in /res.

### DISCUSSION THREAD

**Discussion by ctoabidmaqbool** [iBotPeaches/Apktool#3162] [Ideas]
### Information
1. **Apktool Version (`apktool -version`)** - `2.6.1`
2. **Operating System (Mac, Linux, Windows)** - `Windows 10 64 bit OS`
3. **APK From? (Playstore, ROM, Other)** - `Javafx Mobile Plugin` (`org.javafxports:jfxmobile-plugin:1.3.18`, `gradle apkRelease`)

### Stacktrace/Logcat

```
D:\NetBeansProjects\git-projects\APK2AAB\ApkTool>apktool d eSalesAnalyst2022.10.24.apk -s -o decompile_apk -f
I: Using Apktool 2.6.1 on eSalesAnalyst2022.10.24.apk
I: Loading resource table...
I: Decoding AndroidManifest.xml with resources...
I: Loading resource table from file: C:\Users\MaqboolSoftPC\AppData\Local\apktool\framework\1.apk
I: Regular manifest package...
I: Decoding file-resources...
I: Decoding values */* XMLs...
I: Copying raw classes.dex file...
I: Copying raw classes2.dex file...
I: Copying raw classes3.dex file...
I: Copying assets and libs...
I: Copying unknown files...
I: Copying original files...
```

### Steps to Reproduce
1. `apktool if eSalesAnalyst2022.10.24.apk `
2. `apktool d eSalesAnalyst2022.10.24.apk -s -o decompile_apk -f`
3. `Rename eSalesAnalyst2022.10.24.apk to eSalesAnalyst2022.10.24.zip for comparing result`

### Frameworks
If this APK is from an OEM ROM (Samsung, HTC, LG). Please attach framework files
(`.apks` that live in `/system/framework` or `/system/priv-app`)
**`No any file yet`**

### APK
If this APK can be freely shared, please upload/attach a link to it.
https://drive.google.com/file/d/1gWQbbR8G08cWIYwnyVIhmosTX0FSGGEe/view?usp=share_link

### Questions to ask before submission
1. Have you tried `apktool d`, `apktool b` without changing anything? Yes, all the mentioned res files (e.g. `decompile_apk/res/` ) are removed in final build file
4. If you are trying to install a modified apk, did you resign it? no need, as resources files are missing
5. Are you using the latest apktool version? yes

![image](https://user-images.githubusercontent.com/10474725/199661103-9536e5fc-d750-4ace-964d-b739ef206cb0.png)

* APK file under `decompile_apk\dist` after run command `apktool b decompile_apk`:
https://drive.google.com/file/d/15L_PWO7Qsdfn510pPbkGbgFfR_XHJFIT/view?usp=share_link

**Comment by RenateUSB** [comment]
Those aren't "official" Android res subdirectories.
They really should have put that stuff as res/raw or assets.

**Comment by iBotPeaches** [comment]
I imagine we might be able to fix this, but not the easiest. The problem is we currently iterate the application during disassembly and toss out any file that we know is part of an official Android application - that includes the folder `/res`, so with unknown files dumped in there. We never know they exist.

The problem for this is what do you trust as an unknown file in /res? You parse the entire arsc file for file resources before or after implicit qualifiers are added? So I'll mark this as an enhancement, because I've never seen really any app developer abuse an application and dump things in /res so I imagine this doesn't happen often.

**Comment by ctoabidmaqbool** [comment]
Just a note using javafx-mobile-plugin / gluon mobile plugins, same codding can be used for Desktop (Windows, Linux and Mac), Android and IOS. And so and using other plugins same codding with little bit modification and be ported to web.

So organizing files as of android is not a better option at all.

**Comment by iBotPeaches** [comment]
> Just a note using javafx-mobile-plugin / gluon mobile plugins, same codding can be used for Desktop (Windows, Linux and Mac), Android and IOS. And so and using other plugins same codding with little bit modification and be ported to web.
> 
> So organizing files as of android is not a better option at all.

I'm sorry I have no idea what this means in relation to what I said and Apktool.

**Comment by reubenscratton** [comment]
I have found a Play Store game that contains a Unity .meta file in its /res/xml/.  This is certainly due to an error in the developer's build process.  `apktool d` unpacks it fine but `apktool b` (or rather, `aapt2`) errors out on the non-xml content.

Obviously this is not apktool's fault but it would be nice if it could be more forgiving of developer errors, maybe just warning about unexpected file types.  I can't just delete the .meta file as it is referenced (wrongly) from res/values/public.xml with ".xml" actually in the resource name.

**Reply by IgorEisberg** [reply]
The building process is done with aapt2, developed by Google. Nothing to do with Apktool. Overriding it to be more "forgiving" will introduce a whole lot of problems down the line.
If the file was encoded as a resource, then your only option is to manually remove it.
